test/support/generator/get-company-generator.ts

```diff
@@ -0,0 +1,236 @@
+import { GetCompanyCompaniesHouseMultipleErrorResponse } from '@ukef/helper-modules/companies-house/dto/get-company-companies-house-multiple-error-response.dto';
+import { GetCompanyCompaniesHouseResponse } from '@ukef/helper-modules/companies-house/dto/get-company-companies-house-response.dto';
+import { GetCompanyResponse, Industry } from '@ukef/modules/companies/dto/get-company-response.dto';
+import { SectorIndustryEntity } from '@ukef/modules/sector-industries/entities/sector-industry.entity';
+
+import { AbstractGenerator } from './abstract-generator';
+import { RandomValueGenerator } from './random-value-generator';
+
+export class GetCompanyGenerator extends AbstractGenerator<CompanyValues, GenerateResult, GenerateOptions> {
+  constructor(protected readonly valueGenerator: RandomValueGenerator) {
+    super(valueGenerator);
+  }
+
+  protected generateValues(): CompanyValues {
+    return {
+      companiesHouseRegistrationNumber: this.valueGenerator.stringOfNumericCharacters({ length: 8 }),
+      companyName: this.valueGenerator.sentence({ words: 2 }),
+      buildingName: this.valueGenerator.sentence({ words: 2 }),
+      buildingNumber: this.valueGenerator.nonnegativeInteger({ max: 99 }).toString(),
+      thoroughfareName: this.valueGenerator.sentence({ words: 2 }),
+      locality: this.valueGenerator.word(),
+      postalCode: this.valueGenerator.postcode(),
+      country: this.valueGenerator.word(),
+      sicCodes: [
+        this.valueGenerator.stringOfNumericCharacters({ length: 5 }),
+        this.valueGenerator.stringOfNumericCharacters({ length: 5 }),
+        this.valueGenerator.stringOfNumericCharacters({ length: 5 }),
+        this.valueGenerator.stringOfNumericCharacters({ length: 5 }),
+      ],
+      industryClassNames: [
+        this.valueGenerator.sentence({ words: 10 }),
+        this.valueGenerator.sentence({ words: 10 }),
+        this.valueGenerator.sentence({ words: 10 }),
+        this.valueGenerator.sentence({ words: 10 }),
+      ],
+      industrySectorCode: this.valueGenerator.integer({ min: 1001, max: 1020 }),
+      industrySectorName: this.valueGenerator.sentence({ words: 4 }),
+    };
+  }
+
+  protected transformRawValuesToGeneratedValues(values: CompanyValues[], { registrationNumber }: GenerateOptions): GenerateResult {
+    const [v] = values;
+    const registrationNumberToUse = registrationNumber || v.companiesHouseRegistrationNumber;
+
+    const companiesHousePath = `/company/${registrationNumberToUse}`;
+
+    const mdmPath = `/api/v1/companies?registrationNumber=${registrationNumberToUse}`;
+
+    const randomDateString = () => this.valueGenerator.date().toISOString().split('T')[0];
+    const randomAccountingReferenceDate = this.valueGenerator.date();
+
+    const shuffleArray = (array: any[]) => {
+      for (let i = array.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        [array[i], array[j]] = [array[j], array[i]];
```

Comment:
Object injection sink, recommend changing the approach please.

Reply:
I've read up on object injection sink and I'm not sure how to avoid it in this instance, but I also don't think it's an issue here. My understanding is that using variables as indexes presents a vulnerability when these variables come from user input, so could be e.g., the string `'constructor'`, which could allow a malicious actor to override the `Array` constructor (or indeed any function in the `Array` prototype chain).

However, in this instance, the variables `i` and `j` are not user-supplied, so I don't think there is an actual security risk here.

Reply:
Can we change fro `for` loop to either `forEach` or `for..of` loop please?

Reply:
Happy to do this, but just wondering what the benefit of doing so is? The implementation would be less readable in my opinion:

```
const shuffleArray = <T>(array: Array<T>) => {
      [...Array(array.length).keys()].reverse().slice(0, -1).forEach((i) => {
        const j = Math.floor(Math.random() * (i + 1));
        [array[i], array[j]] = [array[j], array[i]];
      });
};
```

Or:

```
const shuffleArray = <T>(array: Array<T>) => {
      for (let i of [...Array(array.length).keys()].reverse().slice(0, -1)) {
        const j = Math.floor(Math.random() * (i + 1));
        [array[i], array[j]] = [array[j], array[i]];
      }
};
```

Reply:
Happy to go ahead with `for..of` loop please.

Reply:
Changed to `for..of` loop.